plasmapy/particles/decorators.py

```diff
@@ -32,32 +32,34 @@ class _CallableDataDict(TypedDict, total=False):
     allow_custom_particles: bool
     allow_particle_lists: bool
     annotations: dict[str, Any]
-    any_of: Optional[Union[str, Iterable[str]]]
+    any_of: str | Iterable[str] | None
     callable_: Callable[..., Any]
-    exclude: Optional[Union[str, Iterable[str]]]
+    exclude: str | Iterable[str] | None
     parameters_to_process: list[str]
-    require: Optional[Union[str, Iterable[str]]]
+    require: str | Iterable[str] | None
     signature: inspect.Signature
 
 
-_basic_particle_input_annotations = (
+_basic_particle_input_annotations: tuple[type | TypeAlias, ...] = (
     Particle,  # deprecated
     ParticleLike,
     ParticleListLike,
-    Union[ParticleLike, ParticleListLike],
+    ParticleLike | ParticleListLike,
     (Particle, Particle),  # deprecated
 )
 _optional_particle_input_annotations = tuple(
-    Optional[annotation]
-    for annotation in _basic_particle_input_annotations
+    annotation | None
+    # remove [:-1] index in following line when dropping (Particle, Particle)
+    # as a valid annotation
+    for annotation in _basic_particle_input_annotations[:-1]
```

Comment:
This includes some manual additions to this PR. I had intended to put it in a different PR, but got branches mixed up.  I got a mypy error around here.